test/std/algorithms/alg.sorting/alg.binary.search/binary.search/binary_search.pass.cpp

```diff
@@ -0,0 +1,103 @@
+
+// template<ForwardIterator Iter, class T>
+//   constexpr bool      // constexpr after C++17
+//   binary_search(Iter first, Iter last, const T& value);
+
+#include "oneapi_std_test_config.h"
+#include "test_iterators.h"
+#include "checkData.h"
+
+#include <iostream>
+
+#ifdef USE_ONEAPI_STD
+#    include _ONEAPI_STD_TEST_HEADER(algorithm)
+#    include _ONEAPI_STD_TEST_HEADER(iterator)
+namespace s = oneapi_cpp_ns;
+#else
+#    include <algorithm>
+#    include <iterator>
+namespace s = std;
+#endif
+
+#if TEST_DPCPP_BACKEND_PRESENT
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
+constexpr sycl::access::mode sycl_write = sycl::access::mode::write;
+
+template <class Iter, class T>
+bool
+test(Iter first, Iter last, const T& value, bool x)
+{
+    return (s::binary_search(first, last, value) == x);
+}
+
+class KernelBSearchTest1;
+class KernelBSearchTest2;
+class KernelBSearchTest3;
+class KernelBSearchTest4;
+
+template <typename Iter, typename KC>
+void
+kernel_test()
+{
+    sycl::queue deviceQueue = TestUtils::get_test_queue();
+    sycl::cl_bool ret = false;
+    sycl::range<1> numOfItems{1};
+    sycl::buffer<sycl::cl_bool, 1> buffer1(&ret, numOfItems);
+    const unsigned N = 1000;
+    const int M = 10;
+    int host_vbuf[N];
+    int x = 0;
+    for (size_t i = 0; i < N; ++i)
+    {
+        host_vbuf[i] = x;
+        if (++x == M)
+            x = 0;
+    }
+
+    std::sort(host_vbuf, host_vbuf + N);
+    sycl::range<1> host_buffer_sz{N};
+    sycl::buffer<sycl::cl_int, 1> host_data_buffer(host_vbuf, host_buffer_sz);
+    deviceQueue.submit([&](sycl::handler& cgh) {
+        auto ret_access = buffer1.get_access<sycl_write>(cgh);
+        auto data_access = host_data_buffer.get_access<sycl_read>(cgh);
+        cgh.single_task<KC>([=]() {
+            int device_vbuf[N];
+            for (size_t i = 0; i < N; i++)
+                device_vbuf[i] = data_access[i];
+            ret_access[0] = test(Iter(device_vbuf), Iter(device_vbuf + N), 0, true);
+
+            for (int x = 1; x < M; ++x)
+                ret_access[0] &= test(Iter(device_vbuf), Iter(device_vbuf + N), x, true);
+
+            ret_access[0] &= test(Iter(device_vbuf), Iter(device_vbuf + N), -1, false);
+            ret_access[0] &= test(Iter(device_vbuf), Iter(device_vbuf + N), M, false);
+
+            // Simple local array testing
+
+            /* int d[] = {0, 2, 4, 6};
+	    for (int *e = d; e <= d+4; ++e)
+	        for (int y = -1; y <= 7; ++y)
+	           ret_access[0] &= test(d, e, y, (y % 2 ==0) && ((e - d)*2 > y));
+	   */
+        });
+    });
+
+    auto ret_access_host = buffer1.get_access<sycl_read>();
+    TestUtils::exitOnError(ret_access_host[0]);
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+int
+main()
+{
+#if TEST_DPCPP_BACKEND_PRESENT
+    kernel_test<forward_iterator<const int*>, KernelBSearchTest1>();
+    kernel_test<bidirectional_iterator<const int*>, KernelBSearchTest2>();
+    // Enable the test when GPU runtime fix issue
```

Comment:
Usually we use a kind of "broken" macro to temporary exclude a "broken" test case.
A "broken" macro is defined depending on OS/Compiler version/Library version etc

Reply:
Let's check if it works now, and then decide. Random access iterator is the one people will use most, I believe.

Reply:
Checked: it's working OK. So code has been uncommented.